manifests/init.pp

```diff
@@ -30,57 +53,93 @@
   $secret_key  = 'YOUR_KEY',
   $source      = undef,
   $manage_java = true,
+  $collector_host = undef,
+  $collector_port = undef,
+  $windows_temp_dir  = 'C:\users\Administrator\Downloads',
 ) {
   
-  if str2bool("$manage_java") {
-    # Add java dependency, for more information on puppetlabs-java module
-    # see https://github.com/puppetlabs/puppetlabs-java/blob/master/manifests/init.pp
-    class {'java':
-       distribution          => 'jdk',
-       package               => undef, # can be for example 'java-1.8.0-openjdk',
-     }
-  }
+  case $facts['kernel'] {
+    'Linux': {
 
-  case $osfamily {
+      if str2bool("$manage_java") {
+        # Add java dependency, for more information on puppetlabs-java module
+        # see https://github.com/puppetlabs/puppetlabs-java/blob/master/manifests/init.pp
+        class {'java':
+           distribution          => 'jdk',
+           package               => undef, # can be for example 'java-1.8.0-openjdk',
+         }
+      }
 
-    'redhat', 'suse': { 
-      if $source == undef {
-        $pkg_url = 'https://app.takipi.com/app/download?t=rpm&s=puppet' 
-      } else {
-	$pkg_url = $source
-	}
-      package { 'takipi':
-        ensure    => installed,
-        source    => $pkg_url,
-        provider  => rpm,
+      case $osfamily {
+
+        'redhat', 'suse': { 
+          if $source == undef {
+            $pkg_url = 'https://app.takipi.com/app/download?t=rpm&s=puppet' 
+          } else {
+            $pkg_url = $source
+          }
+          package { 'takipi':
+            ensure    => installed,
+            source    => $pkg_url,
+            provider  => rpm,
+          }
+        }
+
+        'debian': { 
+          if $source == undef {
+            $pkg_url = 'https://app.takipi.com/app/download?t=deb&s=puppet' 
+          } else { $pkg_url=$source}
+
+          exec {'fetch takipi':
+            command   => "/usr/bin/wget -q '${pkg_url}' -O /tmp/takipi.deb",
+            creates   => '/tmp/takipi.deb',
+            logoutput => true,
+          } ->
+          package { 'takipi':
+            ensure    => installed,
+            source    => '/tmp/takipi.deb',
+            provider  => dpkg,
+          }
+        }
+
+        default:  { fail("Unrecognized operating system for Takipi installation") }
+      }
+      ->
+      exec { 'install takipi':
+        cwd       => '/opt/takipi/etc',
```

Comment:
i think it's wise to refresh the entry puppet module to only support the agent in a remote collector mode.

Reply:
Will be changed in separate ticket (OO-10091)